text/vm-2.0.md

```diff
@@ -0,0 +1,338 @@
+    fn undefined(&mut self) -> JsResult<'a, JsUndefined>;
+
+    fn empty_object(&mut self) -> JsResult<'a, JsObject>;
+    fn empty_array(&mut self) -> JsResult<'a, JsArray>;
+
+    fn array_buffer(byte_len: u32) -> JsResult<'a, JsArrayBuffer>;
+    fn buffer(byte_len: u32) -> JsResult<'a, JsBuffer>;
+
+    // future: 
+    // fn function(&mut self, /* ??? */) -> JsResult<'a, JsFunction>;
+    // fn array<I>(&mut self, elts: I) -> JsResult<'a, JsArray>
+    //     where I: IntoIterator,
+    //           I::Item: ToJs;
+    // fn object<I>(&mut self, props: I) -> JsResult<'a, JsObject>
+    //     where I: IntoIterator,
+    //           I::Item: (Key, ToJs);
+}
+```
+
+**cx.lock()**
+Freezes the VM by taking out a lock.
+
+**cx.borrow()**
+**cx.borrow_mut()**
+Convenience APIs for locking the VM and borrowing a single value.
+
+**cx.execute_scoped(|mut cx| { … })**
+Execute a callback with a new `Context` instance. Any locally allocated handles created during the body of the callback are only kept alive for the duration of the callback. This method is useful for executing code that may generate temporary data and allowing that temporary data to get freed by the JavaScript garbage collector, especially inside of a loop that may result in large amounts of temporary garbage.
+
+**cx.compute_scoped(|mut cx| { … })**
+Execute a callback with a new `Context` instance and return a JavaScript value. Any locally allocated handles created during the body of the callback are only kept alive for the duration of the callback, except for the result value, which is kept alive for the duration of the outer `Context`. This method is useful for executing code that may generate temporary data and allowing that temporary data to get freed by the JavaScript garbage collector, especially inside of a loop that may result in large amounts of temporary garbage.
+
+**cx.number(f)**
+**cx.boolean(b)**
+**cx.string(s)**
+**cx.null()**
+**cx.undefined()**
+Convenience constructors for primitive value types.
+
+**cx.empty_object()**
+**cx.empty_array()**
+Convenience constructors for vanilla objects and array objects.
+
+**cx.array_buffer(byte_len)**
+Convenience constructor for creating a new `ArrayBuffer` object.
+
+## Implementations of `Context`
+
+The `CallContext`, `ModuleContext`, `TaskContext`, `ExecuteContext`, and `EvaluateContext` types are the five implementations of `Context`. Each type represents a view of the JavaScript virtual machine in a different kind of execution context.
+
+### `CallContext`
+
+A `CallContext` is the execution context of a call into a native Neon function.
+
+```rust
+struct CallContext<'a, T: This> { /* ... */ };
+
+impl<'a, T: This> CallContext<'a, T> {
+    fn len(&self) -> i32;
+    fn argument_opt(&mut self, i: i32) -> Option<Handle<'a, JsValue>>;
+    fn argument<V: Value>(&mut self, i: i32) -> JsResult<'a, V>;
+    fn this(&mut self) -> Handle<'a, T>;
+    fn callee(&mut self) -> Handle<'a, JsFunction>;
```

Comment:
Are we leaving this out of VM 2.0 or removing it later before the 0.2 release?

Reply:
Good catch! I'll remove it from the RFC.